io/plugins/eu.esdihumboldt.hale.io.gml.test/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriterTest.java

```diff
@@ -670,8 +745,34 @@ private IOReport fillFeatureTest(String elementName, URI targetSchema,
 	 * @throws Exception if any error occurs
 	 */
 	private IOReport fillFeatureTest(String elementName, URI targetSchema,
-			Map<List<QName>, Object> values, String testName, String srsName,
-			boolean skipValueTest, boolean expectWriteFail) throws Exception {
+			Map<List<QName>, Object> values, String testName, String srsName, boolean skipValueTest,
+			boolean expectWriteFail) throws Exception {
+		return fillFeatureTest(elementName, targetSchema, values, testName, srsName, skipValueTest,
+				expectWriteFail, null);
+	}
+
+	/**
+	 * Create a feature, fill it with values, write it as GML, validate the GML
+	 * and load the GML file again to compare the loaded values with the ones
+	 * that were written
+	 * 
+	 * @param elementName the element name of the feature type to use, if
+	 *            <code>null</code> a random element will be used
+	 * @param targetSchema the schema to use, the first element will be used for
+	 *            the type of the feature
+	 * @param values the values to set on the feature
+	 * @param testName the name of the test
+	 * @param srsName the SRS name
+	 * @param skipValueTest if the check for equality shall be skipped
+	 * @param expectWriteFail if the GML writing is expected to fail
+	 * @param windingOrderParam winding order parameter or <code>null</code>
+	 * @return the validation report or the GML writing report if writing
+	 *         expected to fail
+	 * @throws Exception if any error occurs
+	 */
+	private IOReport fillFeatureTest(String elementName, URI targetSchema,
+			Map<List<QName>, Object> values, String testName, String srsName, boolean skipValueTest,
+			boolean expectWriteFail, String windingOrderParam) throws Exception {
```

Comment:
As an enum is used for the parameter now, it would be better to use the enum type here, instead of a string.